sql/solutions/lesson4.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: page
+title: Introduction to SQL Solutions
+---
+# Lesson 1 exercises Solutions
+
+## L4.1 Update the age of the person named Luis Marco Polo.
+
+Solution 1:
+First find person's id:
+```SQL
+SELECT *
+  FROM people
+ WHERE name = 'Luis Marco Polo';
+```
+Update age using id number:
+```SQL
+UPDATE people
+   SET age = age + 1
+ WHERE id = 3;
+```
+
+Solution 2:
+Just use their name as identifier to update age. (When could this cause problems?)
+```SQL
+UPDATE people
+   SET age = age + 1
+ WHERE name = 'Luis Marco Polo';
+```
+## L4.2 Increase by 1 the age of all the people who were born in Italy.
+
+```SQL
+UPDATE people
+   SET age = age + 1
+ WHERE country_of_birth = 'IT';
+```
+
+## L4.3 Set all names to lowercase.
+
+```SQL
+UPDATE people
+   SET name = LOWER(name);
+```
+
+## L4.4 Insert a new student.
+
+Solution 1:
+What problems could arise with this solution?
+```SQL
+INSERT INTO people (id, name)
+VALUES (42, 'Valerie Knight');
+```
+
+Solution 2:
+```SQL
+INSERT INTO people (id, name)
+VALUES (
+	   (SELECT id
+          FROM people
+         ORDER BY id DESC
+         LIMIT 1) + 1,
+       'Valerie Knight');
```

Comment:
This inner select solution is not something the students may know at this stage of the tutorial.

Reply:
I put it as solution 2, as an additional option of how this can be done. This is something students can discuss with their coaches, especially as subqueries were broached in the final question of lesson 2.

Solution 1 is also super risky -- to the point of being incorrect actually: which is why I have the discussion question of 'What problems could arise with this solution?'. It's really easy to add in another student which duplicates an existing id.

I could add in another 2 step solution, find last person id number, then add in your new person with that id + 1.  However, if someone else adds to the db in between the two queries, again you will have duplicate ids. 

This might be a problem with this Sqlite db -- most dbs would guard against this being possible.

Reply:
This is discussed at the beginning of lesson 4 -- unfortunately, the people table does allow you to insert a new person with a duplicate id (primary key).